src/MadnetFactory.sol

```diff
@@ -0,0 +1,397 @@
+// SPDX-License-Identifier: MIT-open-group
+pragma solidity  ^0.8.11;
+import "../lib/utils/DeterministicAddress.sol";
+
+interface ifoo {
```

Comment:
Are these dummy contracts being used for something? Maybe moving them to the test folder would be better, or put them in a mock folder?